pytorch_lightning/trainer/data_loading.py

```diff
@@ -223,6 +239,16 @@ def replace_sampler(self, dataloader: DataLoader, sampler, mode: Optional[Runnin
                     f"`{dataloader_cls_name}(dataset, sampler=DistributedSampler(dataset))`."
                 )
 
+        # wrap the `IterableDataset` into a `CaptureIterableDataset` to record sampler states.
+        if _fault_tolerant_enabled() and isinstance(dl_kwargs["dataset"], IterableDataset):
+            initial_seed = dl_kwargs["generator"].initial_seed() if dl_kwargs.get("generator") is not None else None
```

Comment:
The initial seed doens't seem to be used anymore

Reply:
removed